zjit/src/codegen.rs

```diff
@@ -2293,6 +2318,10 @@ fn gen_string_getbyte_fixnum(asm: &mut Assembler, string: Opnd, index: Opnd) ->
     asm_ccall!(asm, rb_str_getbyte, string, index)
 }
 
+fn gen_string_setbyte_fixnum(asm: &mut Assembler, string: Opnd, index: Opnd, value: Opnd) -> Opnd {
+    asm_ccall!(asm, rb_str_setbyte, string, index, value)
```

Comment:
This could use a comment like this which explains why it doesn't need `gen_prepare_non_leaf_call`.

```suggestion
    // rb_str_setbyte is not leaf, but we guard types and index ranges in HIR
    asm_ccall!(asm, rb_str_setbyte, string, index, value)
```